lib/Watching.js

```diff
@@ -166,6 +169,17 @@ class Watching {
 		}
 	}
 
+	suspend() {
+		this.suspended = true;
```

Comment:
You can pause the watcher similar to `_invalidate`.

Reply:
calls to `Watching.watch` and `Watching._invalidate` should break early when suspended.

Reply:
clear `this.invalid = false` to prevent rerun.

Reply:
@sokra 

> You can pause the watcher similar to `_invalidate`.

Isn't `watcher.pause` pauses the watcher itself (it won't get file change events), we only want to pause the compilation, and keep the watcher alive here.


> Also add an test cases for this. There are `test/watchCases`. Maybe you can use them.

Will do thanks for the tips. :)

Reply:
`watcher.pause` will stop the watcher from emitting events, its still active/attached. Calling `watcher.watch` again will resume events and may update watchers if the watched files changed.

Reply:
`watcher.pause` works! And I found it's not necessary to have an `resume` API, because `invalidate` can do the resume work.

PS: test added